src/mscorlib/shared/System/Globalization/TimeSpanParse.cs

```diff
@@ -487,14 +488,41 @@ internal bool SetFailure(ParseFailureKind kind, string resourceKey, object messa
                     case ParseFailureKind.FormatWithParameter:
                         throw new FormatException(SR.Format(message, messageArgument));
 
-                    case ParseFailureKind.Overflow:
-                        throw new OverflowException(message);
-
                     default:
                         Debug.Assert(kind == ParseFailureKind.Format, $"Unexpected failure {kind}");
                         throw new FormatException(message);
                 }
             }
+
```

Comment:
would it make sense to eliminate ParseFailureKind entirely? replacing this method with three representing the last 3 parsefailurekinds.

Reply:
well, Format/FormatWithParameter could be one method with an optional param.